### PR TITLE
add swaggo and prereqs for generating openapi spec file

### PR DESCRIPTION
**Pull request by loadtheaccumulator**
# Description
Adds the swaggo package and updates the Makefile and Dockerfile with new commands.
Includes /api/README.md for overview information.
Adds an api model file with description

FIXES: THEEDGE-3343

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [x] Refactor

### CODE REVIEW

**Review by acosferreira — APPROVED** [review]
lgtm

**Comment by loadtheaccumulator** [comment]
Repository migration is failing, but not necessary a every start.

File: deploy/clowdapp.yaml [hunk 1]
```diff
@@ -46,12 +46,12 @@ objects:
           - -c
           - edge-api-migrate
           inheritEnv: true
-        - name: run-repositories-migration
-          command:
-          - bash
-          - -c
-          - edge-api-migrate-repositories
-          inheritEnv: true
+#        - name: run-repositories-migration
```
Comment (mgold1234) [inline deploy/clowdapp.yaml:49]: should those rows be in # or deleted?

Comment (loadtheaccumulator) [reply deploy/clowdapp.yaml:49]: We'll know on Monday, but need to merge this so API Docs work can begin.
This file will be revisited one more time when we put swaggo into production.

Comment (loadtheaccumulator) [reply deploy/clowdapp.yaml:49]: Actually, they can be deleted. Pushing another PR...

File: api/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+The **@Router** entry is the URL path and HTTP Method for the endpoint. In Edge API, there is only one Router for each path and method.
+
+### **Define Available and Required Fields and Map to @Param**
+
+#### **Create New API-specific Structs**
+
+The structs used throughout the Edge API can contain fields beyond what is necessary for using an endpoint and make the API Docs confusing for development purposes.
+
+To fix this, create an API-specific struct in a new file under the pkg/models directory with **API** appended to the anem of the struct.
```
Comment (mgold1234) [inline api/README.md:51]: what is 'anem'?

File: api/README.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+**// the RHEL for Edge OS** is the description comment that will be read by Swaggo and will appear a description of the field in the Swagger UI form.
+
+
+### **Define Response Options**
+
+The last step in providing definitions to Swaggo is to update the HTTP Response entries in the annotations with all of the response options (e.g., 200, 400, 500) and the corresponding struct that contains the data.
+
+#### **Create New API-specific Structs**
+
+For the purpose of only providing the available and required fields, it might be necessary to create new Response structs.
```
Comment (mgold1234) [inline api/README.md:102]: it is not clear what do you mean about Response structs - can you write here example?

File: api/README.md (hunk 2 above)
Comment (loadtheaccumulator) [reply api/README.md:51]: "name" typo.

File: api/README.md (hunk 3 above)
Comment (loadtheaccumulator) [reply api/README.md:102]: I can. Same as the structs that are used for the current responses, only they might need to be created like the main structs described above.

Comment (loadtheaccumulator) [reply api/README.md:102]: We need to merge both #2104 and #2105 so I can reference the second PR in this one.

Comment (loadtheaccumulator) [reply api/README.md:102]: I updated the README, but it will reference a non-existent file until #2105 is merged.

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**